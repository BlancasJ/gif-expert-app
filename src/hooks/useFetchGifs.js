import { useEffect, useState } from 'react';
import { getGifts } from '../helpers/getGifs';

const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifts(category).then(images => {
      setState({
        data: images,
        loading: false,
      });
    });
  }, [category]);

  return state;
};

export default useFetchGifs;
