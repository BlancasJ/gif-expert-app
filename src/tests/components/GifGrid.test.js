import React from 'react';
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Tests in GifGrid', () => {
  test('should show snapshot', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category='One punch'/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when useFetchGifs return images', () => {
    const gifs = [
      {
        id: 1,
        url: 'https://media4.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=da2f845e2y7qk5or0ob5qgkwccppdh5v5xduh22van8xm9xb&rid=giphy.gif&ct=g',
        title: 'One Punch Man',
      },
      {
        id: 2,
        url: 'https://media4.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=da2f845e2y7qk5or0ob5qgkwccppdh5v5xduh22van8xm9xb&rid=giphy.gif&ct=g',
        title: 'One Punch Man',
      }
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category='One punch'/>);
    //expect(wrapper).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe(false);
    expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);
  });
});