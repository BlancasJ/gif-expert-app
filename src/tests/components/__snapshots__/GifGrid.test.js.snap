// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifGrid should show snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
